apps/web/src/app/[locale]/explore/_components/index.tsx

```diff
@@ -3,11 +3,22 @@ import { ExploreSection } from './explore-section';
 import { ExploreSectionSkeleton } from './explore-section-skeleton';
 import { getChallengesByTagOrDifficulty } from './explore.action';
 import { Footsies } from '~/components/footsies';
+import { redirect } from 'next/navigation';
+import { getAllFlags } from '~/utils/feature-flags';
+import { getServerAuthSession } from '@repo/auth/server';
 
 // CI fails without this
 export const dynamic = 'force-dynamic';
 
 export async function Explore() {
+  // early acces you must be authorized
+  const session = await getServerAuthSession();
+  const flags = await getAllFlags();
+
+  if (!session && flags.enableEarlyAccess) {
+    return redirect('/waitlist');
+  }
```

Comment:
LG, these guards do the trick for now 🚀

Reply:
yeah its "good enough" imo cause its all temp and im lazy haha